portalnet/src/overlay_service.rs

```diff
@@ -3632,6 +3638,20 @@ mod tests {
 
         let found_enr2 = service.find_enr(&node_id_2).unwrap();
         assert_eq!(found_enr2, enr2);
+
+        // Test discovery node address cache
+        let (_, enr3) = generate_random_remote_enr();
+        let node_id_3 = enr3.node_id();
+
+        let node_addr = NodeAddress {
+            enr: enr3.clone(),
+            socket_addr: SocketAddr::V4(enr3.udp4_socket().unwrap()),
+        };
+
+        service.discovery.put_cached_node_addr(node_addr);
+
+        let found_enr3 = service.find_enr(&node_id_3).unwrap();
+        assert_eq!(found_enr3, enr3);
```

Comment:
Should this test should include checking the case that the node id can still be found after the enr has been updated?

Reply:
I don't feel so, the scope of this test is to check if `OverlayService` `find_enr` is looking into the discovery cache. I think updating Enr is out of scope here.